### PR TITLE
fix the fail2ban removal for bitwarden_rs

### PR DESCRIPTION
**Pull request by szaimen**
This is how I would do it

### CODE REVIEW

**Comment by enoch85** [comment]
@szaimen Please have a look at m changes. Thanks! :+1:

File: apps/fail2ban.sh [hunk 1]
```diff
@@ -27,24 +27,23 @@ debug_mode
 root_check
 
 # Check if fail2ban is already installed
-if ! [ -f /etc/fail2ban/filter.d/nextcloud.conf ]
+if ! [ -f /etc/fail2ban/filter.d/nextcloud.conf ] || is_this_installed fail2ban
```
Comment (szaimen) [inline apps/fail2ban.sh:30]: The correct check should be
`if ! [ -f /etc/fail2ban/filter.d/nextcloud.conf ] && ! is_this_installed fail2ban`

File: apps/fail2ban.sh [hunk 2]
```diff
@@ -27,18 +27,28 @@ debug_mode
...
-    rm -Rf /etc/fail2ban/   
+    if ! does_this_docker_exist bitwarden_rs
+    then
+        print_text_in_color "$ICyan" "Unbanning all currently blocked IPs..."
+        fail2ban-client unban --all
+        apt purge fail2ban -y
+        rm -rf /etc/fail2ban
+    else
+        print_text_in_color "$ICyan" "Unbanning all currently blocked IPs..."
+        fail2ban-client unban --all
```
Comment (szaimen) [inline apps/fail2ban.sh:46]: I guess this would also unban all bitwarden_rs banned ip addresses but should most likely be fine...

**Comment by szaimen** [comment]
> @szaimen Please have a look at m changes. Thanks! 👍

Looks good except the first check as commented

**Pull request merged**